packages/radix-ui-themes/src/components/table.tsx

```diff
@@ -56,71 +56,60 @@ TableBody.displayName = 'TableBody';
 
 type TableRowElement = React.ElementRef<'tr'>;
 type TableRowOwnProps = GetPropDefTypes<typeof tableRowPropDefs>;
-interface TableRowProps
-  extends Omit<ComponentPropsWithoutColor<'tr'>, keyof TableRowOwnProps>,
-    TableRowOwnProps {}
+interface TableRowProps extends ComponentPropsWithout<'tr', RemovedProps>, TableRowOwnProps {}
 const TableRow = React.forwardRef<TableRowElement, TableRowProps>((props, forwardedRef) => {
   const { className, ...rowProps } = extractProps(props, tableRowPropDefs);
   return <tr {...rowProps} ref={forwardedRef} className={classNames('rt-TableRow', className)} />;
 });
 TableRow.displayName = 'TableRow';
 
-type TableCellImplElement = React.ElementRef<'td'>;
-type TableCellImplOwnProps = GetPropDefTypes<typeof tableCellPropDefs>;
-interface TableCellImplProps
-  extends Omit<ComponentPropsWithoutColor<'td'>, keyof TableCellImplOwnProps>,
-    TableCellImplOwnProps {
-  tag?: 'td' | 'th';
-}
-const TableCellImpl = React.forwardRef<TableCellImplElement, TableCellImplProps>(
-  (props, forwardedRef) => {
-    const { tag: Tag = 'td', className, ...cellProps } = extractProps(props, tableCellPropDefs);
-    return (
-      <Tag {...cellProps} ref={forwardedRef} className={classNames('rt-TableCell', className)} />
-    );
-  }
-);
-TableCellImpl.displayName = 'TableCellImpl';
```

Comment:
I removed TableCellImpl because it seemed to make the implementation and types more convoluted than it helped